test/testsuites/migrationtests.go

```diff
@@ -103,3 +104,17 @@ func testCatalogSourceConfigsNotDeleted(t *testing.T) {
 	err = test.Global.Client.Get(context.TODO(), types.NamespacedName{Name: helpers.TestCatalogSourceConfigName, Namespace: namespace}, &v2.CatalogSourceConfig{})
 	require.NoError(t, err, fmt.Sprintf("Non datastore or non UI CatalogSourceConfig %s was deleted after migration", helpers.TestCatalogSourceConfigName))
 }
+
+// testInvalidSubscriptionIgnored ensures that after a cluster is migrated
+// any existing subscription that could not be migrated (because it points
+// to a non existent package, for example) is ignored
+func testInvalidSubscriptionIgnored(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "Could not get namespace")
+
+	err = test.Global.Client.Get(context.TODO(), types.NamespacedName{Name: helpers.TestInvalidSubscriptionName, Namespace: namespace}, &olm.Subscription{})
```

Comment:
Shouldn't you check that the catalogsource field is the same to make sure the subscription wasn't updated?

Reply:
https://github.com/operator-framework/operator-marketplace/pull/240/files/c92ce0484dbcf8bbbdd403118787908637ea134b#diff-25b86b22e24668a7c91ff67f35ffd5cdR87